server/src/main/java/org/elasticsearch/gateway/PersistedClusterStateService.java

```diff
@@ -713,12 +713,12 @@ private WriterStats updateMetadata(Metadata previouslyWrittenMetadata, Metadata
             try (DocumentBuffer documentBuffer = allocateBuffer()) {
 
                 final boolean updateGlobalMeta = Metadata.isGlobalStateEquals(previouslyWrittenMetadata, metadata) == false;
-                if (updateGlobalMeta) {
+               // if (updateGlobalMeta) {
                     final Document globalMetadataDocument = makeGlobalMetadataDocument(metadata, documentBuffer);
                     for (MetadataIndexWriter metadataIndexWriter : metadataIndexWriters) {
                         metadataIndexWriter.updateGlobalMetadata(globalMetadataDocument);
                     }
-                }
+             //   }
 
```

Comment:
We use Writable to persist cluster state since https://github.com/crate/crate/pull/16794

Main difference is that toXcontent used to not persist Metadata.indices

Now we can end up with correct cluster state but incorrect persisted state.
Actually not even incorrect, rather unfriendly for "combine bare Metadata and dedicated IndexMetadata" logic.

Always updating global state helps to get rid of old IndexMetadata 
(that happens when Metadata term is changed, for example

This is suboptimal but let us use Writable and keep dumping the whole Metadata state without introducing smth like `contextAwareWriteTo(StreamOutput out, XContentContext ctx)`